### PR TITLE
FE edit profile form

### LINKED ISSUE #23: FE: Profile Settings > Edit Profile
Create the UI for specifically "edit profile" content

- refer to `setup-profile-edit.png` in provided assets
- This task is not responsible for navbar

### PR DESCRIPTION
**Pull request by hodge-jai**
### What this PR does (required):
RESOLVES #23 
- Adds the edit profile form to the front end

### Screenshots / Videos (front-end only):

Screen Shot 2021-11-18 at 9 30 44 AM

### Any information needed to test this feature (required):
- Test responsiveness of form on different screen sizes

### CODE REVIEW

File: client/src/components/EditProfile/ProfileForm/ProfileForm.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
+import { useState, ChangeEvent } from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import Box from '@material-ui/core/Box';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
```
Comment (sundayezeilo) [inline client/src/components/EditProfile/ProfileForm/ProfileForm.tsx]: You imported this package, which should be used for form validation. If not meant to be used here, it should be removed.

**Review by sundayezeilo — COMMENTED** [review]
Looks good! One minor comment though. Also, align the content of your sidebar to match the mockup :smiley:

File: client/src/components/EditProfile/EditMenu.tsx [hunk 2]
```diff
@@ -66,9 +63,11 @@ const EditMenu = (): JSX.Element => {
           <Tab label="Settings" {...a11yProps(5)} />
         </Tabs>
       </Grid>
-      <Grid item xs={6} md={6} lg={6} component={Paper} className={classes.tabComponents}>
+      <Grid component={Paper} className={classes.tabComponents}>
         <TabPanel value={value} index={0}>
-          Edit Profile Placeholder
+          <Box style={{ height: '500px', overflow: 'auto' }}>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/EditMenu.tsx]: Please avoid the usage of inline styles

File: client/src/components/EditProfile/ProfileForm/ProfileForm.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            }}
+            InputProps={{
+              classes: { input: classes.inputs },
+            }}
+            variant="outlined"
+            name="firstName"
+            autoComplete="first-name"
+            autoFocus
+            helperText={touched.firstName ? errors.firstName : ''}
+            error={touched.firstName && Boolean(errors.firstName)}
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfileForm/ProfileForm.tsx]: 👍🏽

File: client/src/components/EditProfile/ProfileForm/ProfileForm.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            name="gender"
+            style={{ width: 200 }}
+            label={<Typography className={classes.label}>Gender</Typography>}
+            value={values.gender}
+            onChange={handleChange}
+            InputLabelProps={{
+              shrink: true,
+            }}
+          >
+            {genders.map((option) => (
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfileForm/ProfileForm.tsx]: 👍🏽

File: client/src/components/EditProfile/ProfileForm/useStyles.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  submit: {
+    margin: theme.spacing(3, 2, 2),
+    padding: 10,
+    width: 160,
+    height: 56,
+    borderRadius: theme.shape.borderRadius,
+    marginTop: 49,
+    fontSize: 16,
+    backgroundColor: '#f14140',
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfileForm/useStyles.ts]: You can set and apply colours using MUI's theme to ensure consistency throughout the app

**Comment by rajivtitus** [comment]
@hodge-jai Good work here! Please try to match the form to the mocks (we're missing the labels), unless there was a reason behind this? 😄

**Comment by hodge-jai** [comment]
Screen Shot 2021-11-22 at 2 40 23 PM

@rajivtitus fixed the labels, please review!

File: client/src/themes/theme.ts [hunk 6]
```diff
@@ -7,6 +7,13 @@ export const theme = createMuiTheme({
     button: {
       fontFamily: '"Roboto"',
     },
+    subtitle1: {
+      color: '#000000',
+      fontWeight: 'bold',
+      fontFamily: "'Roboto'",
```
Comment (rajivtitus) [inline client/src/themes/theme.ts:13]: Please avoid setting the font-family outside of MUI's theme

File: client/src/components/EditProfile/ProfileForm/ProfileForm.tsx [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        gender: '',
+        email: '',
+        address: '',
+        description: '',
+      }}
+      onSubmit={handleSubmit}
+    >
+      {({ handleSubmit, handleChange, values, touched, errors, isSubmitting }) => (
+        <Grid container>
+          <form onSubmit={handleSubmit} className={classes.form} noValidate>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfileForm/ProfileForm.tsx:39]: Please add the form heading as per the mocks - 'Edit Profile'

File: client/src/components/EditProfile/ProfileForm/ProfileForm.tsx [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+            <Grid item className={classes.field}>
+              <Grid className={classes.label}>
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfileForm/ProfileForm.tsx:41]:
```suggestion
              <Grid item className={classes.label}>
```
We always need to specify the type (item/container/item container) of the Grid item. Ditto throughout!

**Review by rajivtitus — APPROVED** [review]
Please address pending comments prior to merging! Thanks

File: client/src/components/EditProfile/ProfileForm/ProfileForm.tsx [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+                <Typography variant="subtitle1" noWrap>
+                  First Name
+                </Typography>
+              </Grid>
+              <TextField
+                id="firstName"
```
Comment (rajivtitus) [inline client/src/components/EditProfile/ProfileForm/ProfileForm.tsx:47]: Optional: You can also add placeholder text to each text field

Eg: 'Enter your first name'

**Comment by rajivtitus** [comment]
@hodge-jai Can we resolve the merge conflicts and merge this in when you get a chance 😄

**Comment by rajivtitus** [comment]
Closing this PR as the changes here were already a part of the Edit Profile Integration.

**Pull request closed without merging**